### PR TITLE
2018 ideas (gRPC Python Team's, at least)

### PR DESCRIPTION
**Pull request by nathanielmanistaatgoogle**
Preview and proofread [this rendering](https://github.com/nathanielmanistaatgoogle/grpc/blob/summer-of-code/summerofcode/ideas.md).

### CODE REVIEW

File: summerofcode/ideas.md [hunk 1]
```diff
@@ -11,37 +11,46 @@ to know the gRPC code and team!
...
 gRPC Ruby/Java:
 
 1. [jRuby](http://jruby.org) support for gRPC. Develop a jRuby wrapper for gRPC based on grpc-java and ensure that it is API compatible with the existing Ruby implementation and passes all tests.
- * **Required skills:** Java programming language, Ruby programming language.
- * **Likely mentors:** [Michael Lumish](https://github.com/murgatroid99), [Eric Anderson](https://github.com/ejona86).
+    * **Required skills:** Java programming language, Ruby programming language.
+    * **Likely mentors:** [Michael Lumish](https://github.com/murgatroid99), [Eric Anderson](https://github.com/ejona86).
 
 
 gRPC Wire Protocol:
```
Comment (hsaliak) [inline summerofcode/ideas.md]: We can drop this, it exists in wireshark already.

File: summerofcode/ideas.md [hunk 2]
```diff
@@ -11,37 +11,46 @@ to know the gRPC code and team!
...
+    * **Required skills:** Python programming language, open source development across multiple repositories and projects.
+    * **Likely mentors:** [Nathaniel Manista](https://github.com/nathanielmanistaatgoogle), [Ken Payson](https://github.com/kpayson64), [Mehrdad Afshari](https://github.com/mehrdada).
+
+1. [Enable building of gRPC Python with Bazel](https://github.com/grpc/grpc/issues/8079). Bazel is the designated replacement for our constellation of crufty build scripts, but it's still under active development itself. Up for a challenge? gRPC Python could easily be the most complex codebase to be built with Bazel.
+    * **Required skills:** Python programming language, Bazel toolchain, Cython, open source development across multiple repositories and projects.
+    * **Likely mentors:** [Nathaniel Manista](https://github.com/nathanielmanistaatgoogle), [Ken Payson](https://github.com/kpayson64), [Mehrdad Afshari](https://github.com/mehrdada).
 
 gRPC Ruby/Java:
```
Comment (hsaliak) [inline summerofcode/ideas.md]: We can drop this one too.

File: summerofcode/ideas.md [hunk 3]
```diff
@@ -11,37 +11,46 @@ to know the gRPC code and team!
...
-1. Develop and test Python 3.5 Support for gRPC. Make necessary changes to port gRPC and package it for supported platforms.
- * **Required skills:** Python programming language, Python 3.5 interpreter.
- * **Likely mentors:** [Nathaniel Manista](https://github.com/nathanielmanistaatgoogle), [Masood Malekghassemi](https://github.com/soltanmm).
+1. Support static type-checking of both gRPC Python itself and of code that uses gRPC Python. No one likes dynamic typing and Python is finally outgrowing it! There are probably errors in the implementation of gRPC Python that [pytype](https://github.com/google/pytype) or [mypy](http://mypy-lang.org/) could detect. There are certainly errors in other code that uses gRPC Python that they could detect.
+    * **Required skills:** Python programming language, open source development across multiple repositories and projects.
+    * **Likely mentors:** [Nathaniel Manista](https://github.com/nathanielmanistaatgoogle), [Ken Payson](https://github.com/kpayson64), [Mehrdad Afshari](https://github.com/mehrdada).
```
Comment (hsaliak) [inline summerofcode/ideas.md]: I want to help out on this project. Please add me as a mentor

File: summerofcode/ideas.md [hunk 4]
```diff
@@ -11,37 +11,46 @@ to know the gRPC code and team!
...
+    * **Required skills:** C programming language, C++ programming language.
+    * **Likely mentors:** [Nathaniel Manista](https://github.com/nathanielmanistaatgoogle), [Nicolas Noble](https://github.com/nicolasnoble).
+
+1. [Make channel-connectivity-watching cancellable](https://github.com/grpc/grpc/issues/3064). Anything worth waiting for is worth cancelling. The fact that channel connectivity is currently poll-based means that clean shutdown of gRPC channels can take as long as the poll interval. No one should have to wait 200ms to garbage-collect an object.
+    * **Required skills:** C programming language, C++ programming language, Python programming language.
+    * **Likely mentors:** [Nathaniel Manista](https://github.com/nathanielmanistaatgoogle), [Vijay Pai](https://github.com/vjpai).
 
 1. Port gRPC to  one of the major BSD platforms ([FreeBSD](https://freebsd.org), [NetBSD](https://netbsd.org), and [OpenBSD](https://openbsd.org)) and create packages for them. Add [kqueue](https://www.freebsd.org/cgi/man.cgi?query=kqueue) support in the process.
```
Comment (hsaliak) [inline summerofcode/ideas.md]: We can get rid of this one

Comment (nathanielmanistaatgoogle) [reply summerofcode/ideas.md]: Done.

File: summerofcode/ideas.md (hunk 3 above)
Comment (nathanielmanistaatgoogle) [reply summerofcode/ideas.md]: Done.

File: summerofcode/ideas.md (hunk 2 above)
Comment (nathanielmanistaatgoogle) [reply summerofcode/ideas.md]: Done.

File: summerofcode/ideas.md (hunk 1 above)
Comment (nathanielmanistaatgoogle) [reply summerofcode/ideas.md]: Done.

**Pull request merged**